src/EllCrv/Models.jl

```diff
@@ -0,0 +1,278 @@
+################################################################################
+#
+#          EllCrv/Models.jl : Different models of elliptic curves
+#
+# This file is part of Hecke.
+#
+# Copyright (c) 2015, 2016: Claus Fieker, Tommy Hofmann
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# (C) 2016 Tommy Hofmann
+# (C) 2016 Robin Ammon
+# (C) 2016 Sofia Brenner
+# (C) 2022 Jeroen Hanselman
+#
+################################################################################
+
+export short_weierstrass_model, is_short_weierstrass_model, simplified_model, 
+is_simplified_model, integral_model, is_integral_model
+
+################################################################################
+#
+# Short Weierstrass model
+#
+################################################################################
+
+@doc Markdown.doc"""
+    short_weierstrass_model(E::EllCrv{fmpq}) -> 
+      (EE::EllCrv, function(EllCrvPt), function(EllCrvPt))
+
+Transform a curve given in long Weierstrass form over QQ to short Weierstrass
+form. Return short form and both transformations for points on the curve;
+first transformation from E (long form) to EE (short form), 
+second transformation is the inverse of this map.
+"""
+function short_weierstrass_model(E::EllCrv)
+  return _short_weierstrass_model(E)
+end
+
+function _short_weierstrass_model(E::EllCrv{T}) where T
+  R = base_field(E)
+  p = characteristic(R)
+
+  if (p == 2) || (p == 3)
```

Comment:
I can wrtite down a short equatino in char 3 I think - I cannot transform all of them, but they do exist. In char 2 I fear that a short equation cannot be smooth...

Reply:
The code for short Weierstrass form was written by students and I simply moved it around. I wrote simplified_model to transform curves of characteristic 2 and 3 into the Weierstrass models that are usually used there. I should probably change that error message and take a look at what Magma does when you try to compute a short Weierstrass form in characteristic 3.